features/upgrade/sdn/ipsec-upgrade.feature

```diff
@@ -0,0 +1,95 @@
+      | name=hello-pod0 |
+    And evaluation of `pod.node_name` is stored in the :worker0 clipboard
+    And a pod becomes ready with labels:
+      | name=hello-pod1 |
+    And evaluation of `pod.node_name` is stored in the :worker1 clipboard
+    And the Internal IP of node "<%= cb.worker1 %>" is stored in the :host_ip clipboard
+    Given I obtain test data file "networking/net_admin_cap_pod.yaml"
+    #Host network pod for running tcpdump on correcpdonding worker node
+    When I run oc create as admin over "net_admin_cap_pod.yaml" replacing paths:
+      | ["spec"]["nodeName"]                                       | <%= cb.worker1 %>  |
+      | ["metadata"]["namespace"]                                  | ipsec-upgrade      |
+      | ["metadata"]["name"]                                       | hostnw-pod-worker1 |
+      | ["metadata"]["labels"]["name"]                             | network-pod1       |
+      | ["spec"]["containers"][0]["securityContext"]["privileged"] | true               |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+       | name=network-pod1 |
+    And evaluation of `pod.name` is stored in the :hostnw_pod_worker1 clipboard
+    #capturing tcpdump for 2 seconds
+    When admin executes on the "<%= cb.hostnw_pod_worker1 %>" pod:
+       | sh | -c | timeout  --preserve-status 2 tcpdump -i <%= cb.default_interface %> esp |
+    Then the step should succeed
+    #Following will confirm pod-pod encryption
+    #Example ESP packet un-encrypted will look like 16:37:16.309297 IP ip-10-0-x-x.us-east-2.compute.internal > ip-10-0-x-x.us-east-2.compute.internal: ESP(spi=0xf50c771c,seq=0xfaad)
+    And the output should match:
+       | <%= cb.worker0 %>.* > <%= cb.worker1 %>.*: ESP |
+    Given I obtain test data file "networking/net_admin_cap_pod.yaml"
+    When I run oc create as admin over "net_admin_cap_pod.yaml" replacing paths:
+       | ["spec"]["nodeName"]           | <%= cb.worker0 %>  |
+       | ["metadata"]["name"]           | hostnw-pod-worker0 |
+       | ["metadata"]["namespace"]      | ipsec-upgrade      |
+       | ["metadata"]["labels"]["name"] | network-pod0       |
+    #Below socat command will send ESP traffic at a length less than 40 between worker 0 and worker1 nodes
+       | ["spec"]["containers"][0]["command"]                       | ["sh", "-c", "socat /dev/random ip-sendto:<%= cb.host_ip %>:<%= cb.protocol %>"] |
+       | ["spec"]["containers"][0]["securityContext"]["privileged"] | true                                                                               |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+       | name=network-pod0 |
+    #Following will confirm node-node encryption
+    When admin executes on the "<%= cb.hostnw_pod_worker1 %>" pod:
+       | sh | -c | timeout  --preserve-status 60 tcpdump -c 2 -i br-ex "esp and less 40" |
+    Then the step should succeed 
+    And the output should not contain "0 packets captured"
```

Comment:
@anuragthehatter I found the created project will not be deleted automatically when script execution finished in the upgrade testing.
I add deletion step at the end for my two open upgrade PR.
Given the "sctp-upgrade" project is deleted

Reply:
@weliang1 non-system projects should be deleted by cucushift at the end of upgrade scenario. Also none of our upgrade tests ensures deletion at the end for the same reason I believe. @asood-rh Does your upgrade include project deletion step?

Reply:
Also let me check on it how cucushift handles it.

Reply:
@anuragthehatter @asood-rh  Our non-upgrade cases will automatically delete all created namespace in the cluster to make the cluster clean. I should say the deletion step in the upgrade cases is a option or suggestion as I tried to do the same way as script check phase did (I need add this in one of my merged PR), it's better to have but not a mandatory.

Reply:
@weliang1 Yep. Also I couldn't find anything in design doc explaining what to be done with projects post upgrade check. .May be it is supposed to be not deleted for debugging later?@zhaozhanqi any idea?

Reply:
Also @liangxia WDYT

Reply:
I don't have a preference for whether or not cleanup those test resources after post upgrade.
As there is a e2e test followed on the upgraded cluster. The feature owners are more clear whether it will have effect on those e2e testings.

Reply:
I did not delete the project in my upgrade scenario.  we can keep it for debugging when this is issue happen I guess.